### PR TITLE
Added i18n missing translations

### PR DESCRIPTION
**Pull request by sgiaccio**

### CODE REVIEW

File: common/i18n/resources/en.js [hunk 1]
```diff
@@ -308,6 +317,17 @@ module.exports = {
...
 
+  dateUtils: {
+    aMomentAgo: 'A moment ago',
+    hour: 'hour',
+    hour_plural: 'hours',
+    day: 'day',
+    day_plural: 'days',
+    week: 'week',
+    week_plural: 'weeks',
+    timeDiff: `{{diff}} $t(dateUtils.{{unit}}, { 'count': 3 }) ago`,
```
Comment (minotogna) [inline common/i18n/resources/en.js]: why count is 3 by default? what about 1 week ago?

File: common/i18n/resources/en.js [hunk 2]
```diff
@@ -308,6 +317,17 @@ module.exports = {
     }
   },
 
+  dateUtils: {
```
Comment (minotogna) [inline common/i18n/resources/en.js]: should this be added under common? e.g. ```common.date...```

File: common/i18n/resources/en.js [hunk 3]
```diff
@@ -117,6 +117,11 @@ module.exports = {
         ascending: 'ascending',
         descending: 'descending',
       },
+      code: 'code',
+      label: 'label',
+      scientific_name: 'Scientific name',
+      file_uuid: 'File uuid',
+      file_name: 'File name',
```
Comment (minotogna) [inline common/i18n/resources/en.js]: why not to add those to surveyForm where all nodeDefs labels are?

File: webapp/loggedin/modules/data/records/components/recordsRow.js [hunk 4]
```diff
@@ -39,10 +42,10 @@ const RecordsRow = props => {
         )
       }
       <div>
-        {getRelativeDate(Record.getDateCreated(record))}
+        {getRelativeDate(i18n, Record.getDateCreated(record))}
       </div>
       <div>
-        {getRelativeDate(Record.getDateModified(record))}
+        {getRelativeDate(i18n, Record.getDateModified(record))}
```
[the remark this answers is not in the document]
Comment (sgiaccio) [reply webapp/loggedin/modules/data/records/components/recordsRow.js]: Ok

File: common/i18n/resources/en.js (hunk 3 above)
Comment (sgiaccio) [reply common/i18n/resources/en.js]: Ok, added a dataVisHeader section in nodeDefs

File: common/i18n/resources/en.js [hunk 5]
```diff
@@ -117,6 +117,11 @@ module.exports = {
         ascending: 'ascending',
         descending: 'descending',
       },
+      code: 'code',
+      label: 'label',
+      scientific_name: 'Scientific name',
```
[the remark this answers is not in the document]
Comment (sgiaccio) [reply common/i18n/resources/en.js]: Right, referring to the other translations

File: common/i18n/resources/en.js (hunk 2 above)
Comment (sgiaccio) [reply common/i18n/resources/en.js]: Ok

File: common/i18n/resources/en.js (hunk 1 above)
Comment (sgiaccio) [reply common/i18n/resources/en.js]: Right

File: common/survey/nodeDef.js [hunk 6]
```diff
@@ -99,7 +99,7 @@ const isPublished = R.propEq(propKeys.published, true)
 
 const getLabel = (nodeDef, lang) => {
   const label = R.path([keys.props, propKeys.labels, lang], nodeDef)
-  return isBlank(label)
+  return isBlank(label) || label === undefined
```
[the remark this answers is not in the document]
Comment (sgiaccio) [reply common/survey/nodeDef.js]: Ok, updated StringUtils.isBlank implementation

File: webapp/loggedin/modules/data/dataVis/dataQuery/components/tableColumn.js [hunk 7]
```diff
@@ -47,38 +50,39 @@ const TableColumn = (props) => {
...
-                            <ProgressBar
-                              className="running progress-bar-striped"
-                              progress={100}
-                              showText={false}/>
-                          </div>
-                        )
-                    }
-                  </div>
-                  : isHeader && noCols > 1
+                isData
```
[the remark this answers is not in the document]
Comment (sgiaccio) [reply webapp/loggedin/modules/data/dataVis/dataQuery/components/tableColumn.js]: Restored the previous formatting

File: common/i18n/resources/en.js [hunk 8]
```diff
@@ -50,6 +50,16 @@ module.exports = {
...
     yes: 'Yes',
+    date: {
+      aMomentAgo: 'A moment ago',
+      hour: 'hour',
+      hour_plural: 'hours',
+      day: 'day',
+      day_plural: 'days',
+      week: 'week',
+      week_plural: 'weeks',
+      timeDiff: `{{diff}} $t(common.date.{{unit}}, { 'count': {{diff}} }) ago`,
```
Comment (minotogna) [inline common/i18n/resources/en.js]: I would call the param diff, count.

File: common/i18n/resources/en.js [hunk 9]
```diff
@@ -267,6 +281,13 @@ module.exports = {
       analysis: 'Analysis',
     },
     confirmNodeDelete: 'Are you sure you want to delete this item?',
+    dataVisHeader: {
```
Comment (minotogna) [inline common/i18n/resources/en.js]: why adding this new param? can't you just use the prefix that we have? e.g. nodeDefTaxon, nodeDefFile?

File: common/stringUtils.js [hunk 10]
```diff
@@ -19,7 +19,7 @@ const contains = (value = '', string = '') => R.includes(toLower(value), toLower
 const isBlank = R.ifElse(
   isString,
   R.pipe(trim, R.isEmpty),
-  R.always(false)
+  R.isNil
 )
```
Comment (minotogna) [inline common/stringUtils.js]: what if you pass a non empty object? will it return true?

File: common/i18n/resources/en.js (hunk 8 above)
Comment (sgiaccio) [reply common/i18n/resources/en.js]: Ok

File: common/stringUtils.js (hunk 10 above)
Comment (sgiaccio) [reply common/stringUtils.js]: It returns false, as before

File: common/i18n/resources/en.js (hunk 9 above)
Comment (sgiaccio) [reply common/i18n/resources/en.js]: TableColumns uses `{i18n.t(`surveyForm.dataVisHeader.${NodeDefTable.extractColName(nodeDef, col)}`)}`. How would you change the prefix depending on the context (file, taxon, code)?

Comment (minotogna) [reply common/i18n/resources/en.js]: you can get it from NodeDef.getType and capitalise the first letter

Comment (sgiaccio) [reply common/i18n/resources/en.js]: Ok

File: webapp/loggedin/modules/data/dataVis/dataQuery/components/tableColumn.js [hunk 11]
```diff
@@ -1,5 +1,9 @@
 import React from 'react'
+import camelize from 'camelize'
 
+import { capitalizeFirstLetter } from '../../../../../../../common/stringUtils'
```
Comment (minotogna) [inline webapp/loggedin/modules/data/dataVis/dataQuery/components/tableColumn.js]: please use ```import StringUtils from '../../../../../../../common/stringUtils'```

Comment (sgiaccio) [reply webapp/loggedin/modules/data/dataVis/dataQuery/components/tableColumn.js]: Done

**Pull request merged**